registration-processor/registration-processor-registration-status-service-impl/src/main/java/io/mosip/registration/processor/status/service/impl/RegistrationStatusServiceImpl.java

```diff
@@ -401,27 +408,63 @@ public void updateRegistrationStatus(InternalRegistrationStatusDto registrationS
 	/**
 	 * Convert entity to dto and get external status.
 	 *
-	 * @param entity
-	 *            the entity
+	 * @param entities
+	 *            the entities
 	 * @return the registration status dto
 	 */
-	private RegistrationStatusDto convertEntityToDtoAndGetExternalStatus(RegistrationStatusEntity entity) {
+	private RegistrationStatusDto convertEntityToDtoAndGetExternalStatus(List<RegistrationStatusEntity> entities) {
 		RegistrationStatusDto registrationStatusDto = new RegistrationStatusDto();
-		registrationStatusDto.setRegistrationId(entity.getRegId());
-		if (entity.getStatusCode() != null) {
-			RegistrationExternalStatusCode registrationExternalStatusCode = regexternalstatusUtil
-					.getExternalStatus(entity);
 
-			String mappedValue = registrationExternalStatusCode.toString();
+		Optional<RegistrationStatusEntity> parentEntity = entities.stream()
+				.filter(e -> mainProcess.stream().anyMatch(name -> name.equals(e.getRegistrationType())))
+				.findFirst();
 
-			registrationStatusDto.setStatusCode(mappedValue);
+		if (parentEntity.isPresent() && parentEntity.get().getStatusCode() != null) {
+			if (parentEntity.get().getStatusCode()
+					.equals(RegistrationStatusCode.PAUSED_FOR_ADDITIONAL_INFO.toString())) {
+				registrationStatusDto = getRegistrationStatusForSubProcess(entities);
+			} else {
+				RegistrationExternalStatusCode registrationExternalStatusCode = regexternalstatusUtil
+						.getExternalStatus(parentEntity.get());
+
+				String mappedValue = registrationExternalStatusCode.toString();
+				registrationStatusDto.setStatusCode(mappedValue);
+
+			}
+			registrationStatusDto.setRegistrationId(parentEntity.get().getRegId());
 		} else {
 			regProcLogger.error(LoggerFileConstant.SESSIONID.toString(), LoggerFileConstant.REGISTRATIONID.toString(),
-					entity.getReferenceRegistrationId(),
+					parentEntity.get().getReferenceRegistrationId(),
 					PlatformErrorMessages.RPR_RGS_REGISTRATION_STATUS_NOT_EXIST.getMessage());
 		}
+
 		return registrationStatusDto;
+	}
+
+	private RegistrationStatusDto getRegistrationStatusForSubProcess(List<RegistrationStatusEntity> entities) {
+
+		RegistrationStatusDto registrationStatusDto = new RegistrationStatusDto();
+
+		RegistrationStatusEntity subProcessEntity = entities.stream().filter(e -> e.getCreateDateTime() != null)
+				.max(Comparator.comparing(RegistrationStatusEntity::getCreateDateTime)).orElse(null);
 
+		if (subProcessEntity != null) {
+			if (subProcessEntity.getStatusCode().equals(RegistrationStatusCode.PROCESSING.toString())
+					|| subProcessEntity.getStatusCode().equals(RegistrationStatusCode.FAILED.toString())
+					|| subProcessEntity.getStatusCode().equals(RegistrationStatusCode.REJECTED.toString())) {
+				registrationStatusDto.setStatusCode(RegistrationExternalStatusCode.PROCESSING.toString());
+			} else {
+				RegistrationExternalStatusCode registrationExternalStatusCode = regexternalstatusUtil
+						.getExternalStatus(subProcessEntity);
+
```

Comment:
Can we have logic of getting the external status in another method in same class ? 

regexternalstatusUtil.getExternalStatus method will get eventually depricated, since it combines both status of packet that will be given to regclient and full registration status.

Ideally RegistrationExternalStatusCode returned from this service should only be as follows and logic should not harcode any stage names or transaction type codes:

PROCESSING, UIN_GENERATED, REJECTED, REREGISTER

Reply:
Mapping between internal and external status:
PROCESSING - PROCESSING
PROCESSED - UIN_GENERATED
REJECTED - REJECTED
FAILED - REREGISTER
REPROCESS_FAILED - REREGISTER
PAUSED, RESUMABLE,REPROCESS - PROCESSING

Reply:
handled this scenario...

Reply:
Just to add the sub process handling.

PROCESSING, PAUSED, RESUMABLE, REPROCESS - PROCESSING
REPROCESS_FAILED,REJECTED,FAILED, PROCESSED - AWAITING INFORMATION
if it is anything other than this return as PROCESSING